server/lib/scheduler/Scheduler.ts

```diff
@@ -254,110 +254,125 @@ class Scheduler {
     return [];
   }
 
+  private getTreesWithHighestPriority(trees) {
+    const highestPriority = Math.min(...trees.map(t => t.schedulingPriority));
+    return trees.filter(t => t.schedulingPriority === highestPriority);
+  }
+
   private async getActionableWorkspacesInOracleMode({
     maybeSuboptimal,
     userId,
   }) {
```

Comment:
If possible view the below diff in VS Code or something else that uses the VS Code diffing algorithm as opposed to the Github diffing algorithm. The actual change is very minimal (20 lines added, 5 removed).